File: fe/lavadero-fe/src/app/core/services/employee.service.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Employee, EmployeeRole } from '../models/models';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {
  private http = inject(HttpClient);

  getEmployees(page: number = 0, size: number = 20): Observable<any> {
    return this.http.get(`${environment.apiUrl}/employees?page=${page}&size=${size}`);
  }

  getActiveEmployees(): Observable<Employee[]> {
    return this.http.get<Employee[]>(`${environment.apiUrl}/employees/active`);
  }

  getEmployeesByRole(role: EmployeeRole): Observable<Employee[]> {
    return this.http.get<Employee[]>(`${environment.apiUrl}/employees/role/${role}`);
  }

  getEmployeeById(id: string): Observable<Employee> {
    return this.http.get<Employee>(`${environment.apiUrl}/employees/${id}`);
  }

  createEmployee(employee: Partial<Employee>): Observable<Employee> {
    return this.http.post<Employee>(`${environment.apiUrl}/employees`, employee);
  }

  updateEmployee(id: string, employee: Partial<Employee>): Observable<Employee> {
    return this.http.put<Employee>(`${environment.apiUrl}/employees/${id}`, employee);
  }

  toggleEmployeeStatus(id: string): Observable<Employee> {
    return this.http.patch<Employee>(`${environment.apiUrl}/employees/${id}/status`, {});
  }

  deleteEmployee(id: string): Observable<void> {
    return this.http.delete<void>(`${environment.apiUrl}/employees/${id}`);
  }
}

File: be/src/main/java/com/lavadero/service/EmployeeService.java
package com.lavadero.service;

import com.lavadero.dto.EmployeeRequest;
import com.lavadero.dto.EmployeeResponse;
import com.lavadero.model.EmployeeRole;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface EmployeeService {
    EmployeeResponse createEmployee(EmployeeRequest request);
    EmployeeResponse updateEmployee(String id, EmployeeRequest request);
    void deleteEmployee(String id);
    EmployeeResponse getEmployeeById(String id);
    Page<EmployeeResponse> getAllEmployees(Pageable pageable);
    List<EmployeeResponse> getActiveEmployees();
    List<EmployeeResponse> getEmployeesByRole(EmployeeRole role);
    EmployeeResponse toggleEmployeeStatus(String id);
}
